src/matrix3.js

```diff
@@ -191,86 +173,56 @@ export default class Matrix3 extends MathArray {
 
   transformVector2(vector, out) {
     out = out || new Vector2();
-    vec2.transformMat4(out, vector, this);
-    // assert(validateVector2(out));
+    vec2.transformMat3(out, vector, this);
+    validateVector(out, 2);
     return out;
   }
 
-  transformVector3(vector, out = new Vector3()) {
+  transformVector3(vector, out) {
     out = out || new Vector3();
-    vec3.transformMat4(out, vector, this);
-    // assert(validateVector3(out));
+    vec3.transformMat3(out, vector, this);
+    validateVector(out, 3);
     return out;
   }
 
-  // transformVector4(vector, out = new Vector4()) {
-  //   out = out || new Vector4();
-  //   vec4.transformMat4(out, vector, this);
-  //   assert(validateVector4(out));
-  //   return out;
-  // }
-
-  // Transforms any 2, 3 or 4 element vector
-  // returns a newly minted Vector2, Vector3 or Vector4
   transformVector(vector, out) {
     switch (vector.length) {
       case 2:
         return this.transformVector2(vector, out);
       case 3:
         return this.transformVector3(vector, out);
-      // case 4:
-      //   return this.transformVector4(vector, out);
       default:
         throw new Error('Illegal vector');
     }
   }
 
   transformDirection(vector, out) {
+    return this._transformVector(vector, out, 0);
```

Comment:
the transformDirection/transformPoint distinction will only apply if this matrix was used with 2D coordinates. We should either cut these functions (recommended for now) or implement them correctly.